gsl_simulation/services/simulation_projet_service.py

```diff
@@ -1,33 +1,128 @@
         new_montant = (
-            simulation_projet.projet.assiette_or_cout_total * Decimal(new_taux) / 100
+            (simulation_projet.projet.assiette_or_cout_total * Decimal(new_taux) / 100)
+            if simulation_projet.projet.assiette_or_cout_total
+            else 0
         )
         new_montant = round(new_montant, 2)
 
         simulation_projet.taux = new_taux
         simulation_projet.montant = new_montant
         simulation_projet.save()
 
+        if simulation_projet.status == SimulationProjet.STATUS_ACCEPTED:
+            return cls._accept_a_simulation_projet(simulation_projet)
+
+        return simulation_projet
+
     @classmethod
     def update_montant(cls, simulation_projet, new_montant: float):
-        new_taux = (
-            Decimal(new_montant)
-            / Decimal(simulation_projet.projet.assiette_or_cout_total)
-        ) * 100
-        new_taux = round(new_taux, 2)
+        new_taux = ProjetService.compute_taux_from_montant(
+            simulation_projet.projet, new_montant
+        )
+
         simulation_projet.taux = new_taux
         simulation_projet.montant = new_montant
         simulation_projet.save()
+
+        if simulation_projet.status == SimulationProjet.STATUS_ACCEPTED:
+            return cls._accept_a_simulation_projet(simulation_projet)
+
+        return simulation_projet
+
+    PROJET_STATUS_TO_SIMULATION_PROJET_STATUS = {
+        Projet.STATUS_ACCEPTED: SimulationProjet.STATUS_ACCEPTED,
+        Projet.STATUS_UNANSWERED: SimulationProjet.STATUS_REFUSED,
+        Projet.STATUS_REFUSED: SimulationProjet.STATUS_REFUSED,
+        Projet.STATUS_PROCESSING: SimulationProjet.STATUS_PROCESSING,
+    }
+
+    # TODO: Tests
+    @classmethod
+    def get_simulation_projet_status(cls, projet: Projet):
+        return cls.PROJET_STATUS_TO_SIMULATION_PROJET_STATUS.get(projet.status)
+
+    @classmethod
+    def _accept_a_simulation_projet(cls, simulation_projet: SimulationProjet):
+        projet = simulation_projet.projet
+        enveloppe = EnveloppeService.get_parent_enveloppe(simulation_projet.enveloppe)
+        projet.accept(montant=simulation_projet.montant, enveloppe=enveloppe)
+        projet.save()
+
+        updated_simulation_projet = SimulationProjet.objects.get(
+            pk=simulation_projet.pk
+        )
+        return updated_simulation_projet
+
+    @classmethod
+    def is_simulation_projet_in_perimetre(
+        cls, simulation_projet: SimulationProjet, perimetre: Perimetre
+    ):
+        projet_arrondissement = simulation_projet.projet.demandeur.arrondissement
+        if perimetre.arrondissement is not None:
+            return perimetre.arrondissement.pk == projet_arrondissement.pk
+        if perimetre.departement is not None:
+            return perimetre.departement.pk == projet_arrondissement.departement.pk
+        return perimetre.region.pk == projet_arrondissement.departement.region.pk
```

Comment:
Pour l'instant on a dénormalisé : on a aussi le département sur le demandeur, et il est rempli de manière plus systématique. Je propose donc ceci : 

```suggestion
            return perimetre.departement_id == projet.demandeur.departement_id
        return perimetre.region_id == projet.demandeur.departement.region_id
```

En outre j'ai réduit le nombre de `.` pour limiter le nombre de jointures/requêtes nécessaires pour le contrôle, si tu regardes bien !

Reply:
(normalement le passage de `.departement.pk` à `.departement_id` n'est qu'un détail d'implémentation qui ne change pas le résultat des tests, j'ai juste un doute sur les suffixes des champs : est-ce que c'est `_id` ou est-ce que c'est `_code`, par exemple…)